File: tests/algorithms/x/PHP/data_structures/linked_list/is_palindrome.php
<?php
ini_set('memory_limit', '-1');
function _append($arr, $x) {
    $arr[] = $x;
    return $arr;
}
function is_palindrome($values) {
  $stack = [];
  $fast = 0;
  $slow = 0;
  $n = count($values);
  while ($fast < $n && $fast + 1 < $n) {
  $stack = _append($stack, $values[$slow]);
  $slow = $slow + 1;
  $fast = $fast + 2;
};
  if ($fast == $n - 1) {
  $slow = $slow + 1;
}
  $i = count($stack) - 1;
  while ($slow < $n) {
  if ($stack[$i] != $values[$slow]) {
  return false;
}
  $i = $i - 1;
  $slow = $slow + 1;
};
  return true;
}
function main() {
  echo rtrim(json_encode(is_palindrome([]), 1344)), PHP_EOL;
  echo rtrim(json_encode(is_palindrome([1]), 1344)), PHP_EOL;
  echo rtrim(json_encode(is_palindrome([1, 2]), 1344)), PHP_EOL;
  echo rtrim(json_encode(is_palindrome([1, 2, 1]), 1344)), PHP_EOL;
  echo rtrim(json_encode(is_palindrome([1, 2, 2, 1]), 1344)), PHP_EOL;
}
main();

File: tests/algorithms/x/Java/data_structures/linked_list/is_palindrome.java
public class Main {

    static boolean is_palindrome(int[] values) {
        int[] stack = ((int[])(new int[]{}));
        int fast = 0;
        int slow = 0;
        int n = values.length;
        while (fast < n && fast + 1 < n) {
            stack = ((int[])(java.util.stream.IntStream.concat(java.util.Arrays.stream(stack), java.util.stream.IntStream.of(values[slow])).toArray()));
            slow = slow + 1;
            fast = fast + 2;
        }
        if (fast == n - 1) {
            slow = slow + 1;
        }
        int i = stack.length - 1;
        while (slow < n) {
            if (stack[i] != values[slow]) {
                return false;
            }
            i = i - 1;
            slow = slow + 1;
        }
        return true;
    }

    static void main() {
        System.out.println(is_palindrome(((int[])(new int[]{}))));
        System.out.println(is_palindrome(((int[])(new int[]{1}))));
        System.out.println(is_palindrome(((int[])(new int[]{1, 2}))));
        System.out.println(is_palindrome(((int[])(new int[]{1, 2, 1}))));
        System.out.println(is_palindrome(((int[])(new int[]{1, 2, 2, 1}))));
    }
    public static void main(String[] args) {
        main();
    }
}
